syntax = "proto3";

package auth;

service AuthService {
  rpc GenerateTokens (TokenRequest) returns (TokenResponse) {}
}

message TokenRequest {
  string user_id = 1;
  string username = 2;
  string email = 3;
  repeated string roles = 4;
  map<string, string> additional_data = 5;
}

message TokenResponse {
  string access_token = 1 ;
  string token_type = 2;
  int32 expires_in = 3;
  string refresh_token = 4;
}